Graph/Traversal/BreadthFirstSearch.cpp

```diff
@@ -0,0 +1,57 @@
+/* Breadth-first search/traversal
+ * iterates through all vertices in an undirected graph in breadth-first order
+ *
+ * Time-complexity: 
+ * O(V + E), where V is number of vertices and E is number of edges
+ *
+ * Space-complexity:
+ * O(V)
+ *
+ */
+
+#include <queue>
+#include "BreadthFirstSearch.h"
+
+void Graph::breadthFirstSearch(int source_vertex) {
+    bool visited[num_vertices] = { false }; /* tracks which vertices have been visited */
+    std::queue<int> toVisit; /* tracks which vertices to visit next */
+
+    visited[source_vertex] = true;
+    toVisit.push(source_vertex);
+
+    int curr_vertex;
+    while (!toVisit.empty()) {
+        curr_vertex = toVisit.front(); /* current vertex being visited */
+        std::cout << curr_vertex << " ";
+        
+        /* visits all the adjacent vertices and pushes them onto the queue */
+        for (auto itr = adj_list[curr_vertex].begin(); itr != adj_list[curr_vertex].end(); itr++) {
```

Comment:
It would be better to change this to a _range-based for loop_ if that improves simplicity and readability.